code/modules/reagents/chemistry/machinery/chem_recipe_debug.dm

```diff
@@ -0,0 +1,112 @@
+/*
+* A debug chem tester that will process through all recipies automatically and try to react them.
+* Highlights low purity reactions and and reactions that don't happen
+*/
+/obj/machinery/chem_recipe_debug
+	name = "chemical reaction tester"
+	density = TRUE
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "HPLC"
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 40
+	resistance_flags = FIRE_PROOF | ACID_PROOF | INDESTRUCTIBLE
+	///List of every reaction in the game kept locally for easy access
+	var/list/cached_reactions = list()
```

Comment:
If this list won't change, then consider making it a static or global-managed variable.
If you are doing this for optimization purposes, to quickly access the list, you can do something like this:
```
/proc/whose_list_access_is_optimized()
  var/list/local_variable = GLOB.global_list_pointer_of_which_is_now_copied
```

Reply:
There is already a `GLOB.chemical_reactions_list` that is indexed by reagents respective to reactions. This differs in that it contains a list of all in the `datum/chemical_reaction` typepath so that it only does a single reaction once.

Reply:
The `chem_recipe_debug` is for testing purposes only, not for the actual game, right?
If so feel resolve this issue, anything is fine here, really.

Reply:
Yes, it should never appear in game, if it does it will force every reaction to happen on the tile/machine, and this results in the area around it exploding a lot.